src/redux/history/methods.ts

```diff
@@ -106,15 +106,15 @@ export const toggleHistoryEntryPinning = async (noteId: string): Promise<void> =
   if (!entryToUpdate) {
     return Promise.reject(`History entry for note '${noteId}' not found`)
   }
-  if (entryToUpdate.pinStatus === undefined) {
-    entryToUpdate.pinStatus = false
+  const updatedEntry = {
+    ...entryToUpdate,
+    pinStatus: !entryToUpdate.pinStatus
```

Comment:
This uses the casting of undefined to false, is pinStatus now just not able to be undefined?

Reply:
The backend DTO clearly defines a boolean as type.
https://github.com/hedgedoc/hedgedoc/blob/develop/src/history/history-entry.dto.ts#L49

The HistoryEntry in fact also only allows a boolean as type, therefore the if-condition seems to be an unnecessary relict from the past.
https://github.com/hedgedoc/react-client/blob/alpha-fixes/src/api/history/types.ts#L22